app/services/membership/withdraw_service.rb

```diff
@@ -0,0 +1,29 @@
+# frozen_string_literal: true
+
+class Membership::WithdrawService
+  attr_reader :group, :user
+
+  class MembershipMustHaveDefaultPost < StandardError; end
+
+  def initialize(group, user)
+    @group = group
+    @user = user
+  end
+
+  def self.call(group, user)
+    new(group, user).perform
+  end
+
+  def perform
+    membership = user.membership_for(group)
+    raise MembershipMustHaveDefaultPost unless membership.has_post?(PostType::DEFAULT_POST_ID)
+
+    notifications_from_user_to_group = group.leader.user
+                                                   .notifications.where(notifier_id: @user.id)
+                                                   .select{ |n| n.notifiable.group == @group }
+    membership.destroy!
+
+    invalid_notification = notifications_from_user_to_group.find{ |n| !n.reload.notifiable.present? }
+    invalid_notification.destroy!
```

Comment:
Yes, it leaves an empty notification by default. So If you apply/withdraw multiple times, you can spam the leader with notifications. I didn't find a solution in ActivityNotifier's documentation (it's possible that just I missed it, please take a look).
I wrote a query to delete the notification for the withdrawn membership.
It is messy because I had to be very specific which notification needs to be deleted.
Is it possible for `notifiable.present? ` be `false` when the membership not withdrawn?
If yes. then all of those notifications should be deleted? 
I used `find` for getting the notification for deletion, which returns only the first match, so it would be problem if `invalid_notification` could be an array.